src/bin/download_sysext.rs

```diff
@@ -260,10 +262,10 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let client = reqwest::Client::new();
 
     for pkg in pkgs_to_dl.iter_mut() {
-        pkg.check_download(&unverified_dir)?;
-
         pkg.download(&unverified_dir, &client).await?;
 
+        pkg.check_download(&unverified_dir)?;
```

Comment:
I think the idea was to reuse a previous download. If too complicated we can also remove it. Now it calculates the checksums twice which doesn't make sense:
```
[2023-11-03T10:15:35Z DEBUG download_sysext]     expected sha256:   46a5f66202f89f9ca93b90476466bf8325976749a22aac2550e84cc4e3d61bb7
[2023-11-03T10:15:35Z DEBUG download_sysext]     calculated sha256: 46a5f66202f89f9ca93b90476466bf8325976749a22aac2550e84cc4e3d61bb7
[2023-11-03T10:15:35Z DEBUG download_sysext]     sha256 match?      true
[2023-11-03T10:15:35Z INFO  download_sysext] /var/tmp/outdir/.unverified/oem-ami.gz: download complete, checking hash...
[2023-11-03T10:15:36Z DEBUG download_sysext]     expected sha256:   46a5f66202f89f9ca93b90476466bf8325976749a22aac2550e84cc4e3d61bb7
[2023-11-03T10:15:36Z DEBUG download_sysext]     calculated sha256: 46a5f66202f89f9ca93b90476466bf8325976749a22aac2550e84cc4e3d61bb7
[2023-11-03T10:15:36Z DEBUG download_sysext]     sha256 match?      true
```

Reply:
Right. Instead I have added checking for a missing file in `check_download`.